File: script/tests/test_fill_wispertalk.py
import os
import sqlite3
import subprocess
import tempfile
from pathlib import Path


def test_fill_wispertalk() -> None:
    with tempfile.TemporaryDirectory() as tmpdir:
        db_path = Path(tmpdir) / "db.sqlite"
        conn = sqlite3.connect(db_path)
        conn.execute(
            """
            CREATE TABLE sales_call_ratings (
                id INTEGER PRIMARY KEY AUTOINCREMENT,
                filepath TEXT,
                WisperTALK TEXT
            )
            """
        )
        conn.execute(
            "INSERT INTO sales_call_ratings (filepath) VALUES ('dummy.wav')"
        )
        conn.commit()
        conn.close()

        env = os.environ.copy()
        env.update({
            "DB_CONNECTION": "sqlite",
            "DB_DATABASE": str(db_path),
        })

        subprocess.run(
            ["php", os.path.join("script", "tests", "test_fill_wispertalk.php")],
            check=True,
            env=env,
        )

        conn = sqlite3.connect(db_path)
        text = conn.execute(
            "SELECT WisperTALK FROM sales_call_ratings"
        ).fetchone()[0]
        conn.close()
        assert text == "stub text"

File: script/tests/test_fill_wispertalk.php
<?php
require_once __DIR__ . '/../../app/Support/WisperTalk.php';

$database = getenv('DB_DATABASE') ?: ':memory:';
$dsn = 'sqlite:' . $database;
$pdo = new PDO($dsn);
$pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);

$pdo->exec("CREATE TABLE IF NOT EXISTS sales_call_ratings (
    id INTEGER PRIMARY KEY AUTOINCREMENT,
    filepath TEXT,
    WisperTALK TEXT,
    Dept TEXT
)");
$pdo->exec("INSERT INTO sales_call_ratings (filepath, Dept) VALUES ('dummy.wav', 'Sales')");

process_missing_transcriptions($pdo, function (string $path): string {
    return 'stub text';
});

$text = $pdo->query('SELECT WisperTALK FROM sales_call_ratings')->fetchColumn();
if ($text !== 'stub text') {
    fwrite(STDERR, "Test failed\n");
    exit(1);
}
